internal/pkg/initialize/workload.go

```diff
@@ -307,39 +307,41 @@ func (w *WorkloadInitializer) newServiceManifest(i *ServiceProps) (encoding.Bina
 	}
 }
 
-func (w *WorkloadInitializer) newLoadBalancedWebServiceManifest(i *ServiceProps) (*manifest.LoadBalancedWebService, error) {
+func (w *WorkloadInitializer) newLoadBalancedWebServiceManifest(inProps *ServiceProps) (*manifest.LoadBalancedWebService, error) {
 	var httpVersion string
-	if i.Port == commonGRPCPort {
+	if inProps.Port == commonGRPCPort {
 		log.Infof("Detected port %s, setting HTTP protocol version to %s in the manifest.\n",
-			color.HighlightUserInput(strconv.Itoa(int(i.Port))), color.HighlightCode(manifest.GRPCProtocol))
+			color.HighlightUserInput(strconv.Itoa(int(inProps.Port))), color.HighlightCode(manifest.GRPCProtocol))
 		httpVersion = manifest.GRPCProtocol
 	}
-	props := &manifest.LoadBalancedWebServiceProps{
+	outProps := &manifest.LoadBalancedWebServiceProps{
 		WorkloadProps: &manifest.WorkloadProps{
-			Name:                    i.Name,
-			Dockerfile:              i.DockerfilePath,
-			Image:                   i.Image,
-			PrivateOnlyEnvironments: i.PrivateOnlyEnvironments,
+			Name:                    inProps.Name,
+			Dockerfile:              inProps.DockerfilePath,
+			Image:                   inProps.Image,
+			PrivateOnlyEnvironments: inProps.PrivateOnlyEnvironments,
 		},
 		Path:        "/",
-		Port:        i.Port,
+		Port:        inProps.Port,
 		HTTPVersion: httpVersion,
-		HealthCheck: i.HealthCheck,
-		Platform:    i.Platform,
+		HealthCheck: inProps.HealthCheck,
+		Platform:    inProps.Platform,
 	}
-	existingSvcs, err := w.Store.ListServices(i.App)
+	existingSvcs, err := w.Store.ListServices(inProps.App)
 	if err != nil {
 		return nil, err
 	}
-	// We default to "/" for the first service, but if there's another
+	// We default to "/" for the first service or if the application is initialized with a domain, but if there's another
```

Comment:
may be a dumb question: If an application is initialized with domain, does this mean for all the services in that application will have `http.path: '/'`

Reply:
When an app is initialized with a domain, say `example.com`, then we can use host-header based routing for LoadBalancedWebServices.

So for example, `frontend.test.example.com/` will route to the `frontend` service; whereas, `backend.test.example.com/` will route to the `backend` service.

Whereas if there is no domain that's associated with the application, then we have to use path based routing.
So for example, `http://alb-demo-1234567890.us-west-2.elb.amazonaws.com/` will go to the `frontend` service; and since we cannot use the same URL to navigate to a different service we give instead the default path: `http://alb-demo-1234567890.us-west-2.elb.amazonaws.com/backend` to go to `backend`.

Hope that clarified how things worked behind the scenes :)